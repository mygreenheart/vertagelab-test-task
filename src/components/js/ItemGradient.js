import React from 'react'
import { Card, Row, Col, Button } from "react-bootstrap"
import "../css/style.css"


export default function ItemGradient(props) {


    return (
        <Card style={{ width: '80%' }} className="mb-2" className="gradient-item bg-dark text-white text-center">
            <Card.Header>
                <Row>
                    <Col>
                        <Button href={`/edit/${props.id}`} variant="outline-primary" size="sm">Edit</Button>
                    </Col>
                    <Col>
                        <Button variant="outline-danger" size="sm" onClick={() => props.handleDelete(props.id)}>Delete</Button>
                    </Col>
                </Row>
            </Card.Header>
            <Card.Body style={{ background: `linear-gradient(to left, ${props.colorTo},${props.colorFrom})` }}>
                <Row>
                    <Col>{props.colorFrom}</Col>
                    <Col>{props.colorTo}</Col>
                </Row>
            </Card.Body>
        </Card>
    )
}