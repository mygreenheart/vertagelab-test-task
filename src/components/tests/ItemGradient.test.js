import React from 'react';
import ItemGradient from "../js/ItemGradient"

import Enzyme, { shallow, mount } from 'enzyme';
import { render, fireEvent, cleanup } from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

const setUp = (props) => shallow(<ItemGradient {...props} />)

describe('Item gradient js', () => {
    let component
    let instance
    beforeEach(() => {
        component = setUp()
        instance = component.instance()
    })

    it("shoude correct render", () => {
        const wrapper = shallow(<ItemGradient />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

});

