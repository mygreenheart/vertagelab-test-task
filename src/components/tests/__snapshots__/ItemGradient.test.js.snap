// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item gradient js shoude correct render 1`] = `
<Card
  body={false}
  className="gradient-item bg-dark text-white text-center"
  style={
    Object {
      "width": "80%",
    }
  }
>
  <CardHeader>
    <Row
      noGutters={false}
    >
      <Col>
        <Button
          active={false}
          disabled={false}
          href="/edit/undefined"
          size="sm"
          variant="outline-primary"
        >
          Edit
        </Button>
      </Col>
      <Col>
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          size="sm"
          variant="outline-danger"
        >
          Delete
        </Button>
      </Col>
    </Row>
  </CardHeader>
  <CardBody
    style={
      Object {
        "background": "linear-gradient(to left, undefined,undefined)",
      }
    }
  >
    <Row
      noGutters={false}
    >
      <Col />
      <Col />
    </Row>
  </CardBody>
</Card>
`;
